modules/infra/control/loopback.c

```diff
@@ -0,0 +1,196 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024 Christophe Fontaine
+
+#include <gr.h>
+#include <gr_control_input.h>
+#include <gr_control_output.h>
+#include <gr_eth.h>
+#include <gr_iface.h>
+#include <gr_infra.h>
+#include <gr_log.h>
+#include <gr_loopback.h>
+#include <gr_mempool.h>
+#include <gr_module.h>
+
+#include <event2/event.h>
+
+#include <fcntl.h>
+#include <linux/if_tun.h>
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#define TUN_TAP_DEV_PATH "/dev/net/tun"
+
+static struct rte_mempool *loopback_pool;
+extern control_input_t control_to_loop_input;
+static struct event_base *global_ev_base;
+
+struct iface_info_loop {
+	int fd;
+	struct event *ev;
+};
+
+void loopback_tx(struct rte_mbuf *m) {
+	struct mbuf_data *d = mbuf_data(m);
+	struct iface_info_loop *loop;
+
+	loop = (struct iface_info_loop *)d->iface->info;
+	if (write(loop->fd, rte_pktmbuf_mtod(m, char *), m->data_len) < 0)
```

Comment:
@david-marchand is this correct to use `m->data_len` here? Think it would be better to use `rte_pktmbuf_pkt_len(m)`.

Reply:
Well, it is not clear what is expected here.
rte_pktmbuf_pkt_len() would be incorrect, since write() expects a single linear buffer.

If m is a multi segment packet, then we have a problem.

Reply:
Thanks ! Will add that to the next version.